addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/environment-resource-service.js

```diff
@@ -0,0 +1,287 @@
+      policyDoc.addStudyRole(roleArn);
+    });
+
+    return policyDoc;
+  }
+
+  /**
+   * Populate s3 mount information that is going to be passed to the user data of the compute instance. This s3 mount
+   * information is then used by the mount script that runs on the compute instance. It is important to keep in mind
+   * that other study access strategy might be participating in this logic.
+   *
+   * @param requestContext The standard request context object
+   * @param studies an array of StudyEntity that are associated with this env. IMPORTANT: each element in the array is
+   * the standard StudyEntity, however, there is one additional attributes added to each of the StudyEntity. This
+   * additional attribute is called 'envPermission', it is an object with the following shape:
+   * { read: true/false, write: true/false }
+   * @param s3Mounts An array that contains the s3 mount information. Each element in the array is an object with the
+   * following shape: { id, bucket, region, awsPartition, kmsArn, prefix, roleArn, readable, writable }
+   * @param environmentScEntity the EnvironmentSc entity. The entity is expected to have an attributed named 'studyRoles'.
+   * The studyRoles is a map where the key is the study id and the value is the fs role arn
+   */
+  async provideStudyMount(requestContext, { studies: allStudies, s3Mounts, environmentScEntity }) {
+    // We are only interested in studies with bucket access = 'roles'
+    const studies = _.filter(allStudies, study => study.bucketAccess === 'roles');
+
+    if (_.isEmpty(studies)) return s3Mounts; // No relevant studies for this access strategy, so we are done
+
+    // We loop through all the studies and add items to the s3 mount information array
+    _.forEach(studies, study => {
+      const { id, bucket, kmsScope, folder, region, awsPartition, envPermission = {} } = study;
+      const { read, write } = envPermission;
+      // The logic to determining the kmsArn logic is:
+      // - If the study kms scope is 'bucket', then we don't include the kmsArn in the mount information, this way
+      //   the default bucket kms will be used by S3 without needing to pass the exact kms arn
+      // - If the study kms scope is 'study', then we need to include the kmsArn in the mount information
+      const kmsArn = kmsScope === 'study' ? study.kmsArn : undefined;
+      const roleArn = _.get(environmentScEntity, 'studyRoles', {})[id];
+      const item = { id, bucket, region, kmsArn, roleArn, prefix: folder, readable: read, writeable: write };
+      if (awsPartition !== 'aws') item.awsPartition = awsPartition;
+
+      s3Mounts.push(item);
+    });
+
+    return s3Mounts;
+  }
+
+  // @private
+  // Returns a map. The key is the bucket name and the value is the bucket kmsArn. IMPORTANT: not all studies buckets
+  // will be included in the map. Only Buckets for studies with kmsScope = bucket are included.
+  async getBuckets(studies) {
```

Comment:
Where is this method used? Also perhaps rename it to getKMSScopedBuckets or something

Reply:
Sorry, it was used by code that is no longer in the file.  I removed the method just now.